### DISCUSSION TITLE
error while runing locally

### DISCUSSION THREAD

**Discussion by AdityaRaimec22** [scorelab/Codelabz#920] [Q&A]
codelabz-app-1       |
codelabz-app-1       | > codelabz@0.0.0 dev /app
codelabz-app-1       | > vite
codelabz-app-1       |
codelabz-app-1       | sh: 1: vite: not found
codelabz-app-1       | npm ERR! code ELIFECYCLE
codelabz-app-1       | npm ERR! syscall spawn
codelabz-app-1       | npm ERR! file sh
codelabz-app-1       | npm ERR! errno ENOENT
codelabz-app-1       | npm ERR! codelabz@0.0.0 dev: `vite`
codelabz-app-1       | npm ERR! spawn ENOENT
codelabz-app-1       | npm ERR!
codelabz-app-1       | npm ERR! Failed at the codelabz@0.0.0 dev script.
codelabz-app-1       | npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
codelabz-app-1       | npm WARN Local package.json exists, but node_modules missing, did you mean to install?
codelabz-app-1       |
codelabz-app-1       | npm ERR! A complete log of this run can be found in:
codelabz-app-1       | npm ERR!     /root/.npm/_logs/2023-10-22T01_37_34_503Z-debug.log
codelabz-app-1 exited with code 1
codelabz-emulator-1  | ⚠  emulators: You are not currently authenticated so some features may not work correctly. Please run firebase login to authenticate the CLI.
codelabz-emulator-1  | i  emulators: Starting emulators: auth, functions, firestore, database, pubsub, storage
codelabz-emulator-1  | i  emulators: Detected demo project ID "demo-sampark", emulated services will use a demo configuration and attempts to access non-emulated services for this project will fail.
codelabz-emulator-1  | ⚠  emulators: It seems that you are running multiple instances of the emulator suite for project demo-sampark. This may result in unexpected behavior.
codelabz-emulator-1  | ⚠  emulators: Could not find import/export metadata file, skipping data import!
codelabz-emulator-1  | ⚠  functions: The functions emulator is configured but there is no functions source directory. Have you run firebase init functions?
codelabz-emulator-1  | ⚠  firestore: Did not find a Cloud Firestore rules file specified in a firebase.json config file.
codelabz-emulator-1  | ⚠  firestore: The emulator will default to allowing all reads and writes. Learn more about this option: https://firebase.google.com/docs/emulator-suite/install_and_configure#security_rules_configuration.
codelabz-emulator-1  | i  firestore: Firestore Emulator logging to firestore-debug.log
codelabz-emulator-1  | ⚠  database: Did not find a Realtime Database rules file specified in a firebase.json config file. The emulator will default to allowing all reads and writes. Learn more about this option: https://firebase.google.com/docs/emulator-suite/install_and_configure#security_rules_configuration.
codelabz-emulator-1  | i  database: Database Emulator logging to database-debug.log
codelabz-emulator-1  | i  pubsub: Pub/Sub Emulator logging to pubsub-debug.log
codelabz-emulator-1  | i  ui: Emulator UI logging to ui-debug.log
codelabz-emulator-1  |
codelabz-emulator-1  | ┌─────────────────────────────────────────────────────────────┐
codelabz-emulator-1  | │ ✔  All emulators ready! It is now safe to connect your app. │
codelabz-emulator-1  | │ i  View Emulator UI at http://0.0.0.0:4000                  │
codelabz-emulator-1  | └─────────────────────────────────────────────────────────────┘
codelabz-emulator-1  |
codelabz-emulator-1  | ┌────────────────┬──────────────────────────────────┬───────────────────────────────┐
codelabz-emulator-1  | │ Emulator       │ Host:Port                        │ View in Emulator UI           │
codelabz-emulator-1  | ├────────────────┼──────────────────────────────────┼───────────────────────────────┤
codelabz-emulator-1  | │ Authentication │ 0.0.0.0:9099                     │ http://0.0.0.0:4000/auth      │
codelabz-emulator-1  | ├────────────────┼──────────────────────────────────┼───────────────────────────────┤
codelabz-emulator-1  | │ Functions      │ Failed to initialize (see above) │                               │
codelabz-emulator-1  | ├────────────────┼──────────────────────────────────┼───────────────────────────────┤
codelabz-emulator-1  | │ Firestore      │ 0.0.0.0:8080                     │ http://0.0.0.0:4000/firestore │
codelabz-emulator-1  | ├────────────────┼──────────────────────────────────┼───────────────────────────────┤
codelabz-emulator-1  | │ Database       │ 0.0.0.0:9000                     │ http://0.0.0.0:4000/database  │
codelabz-emulator-1  | ├────────────────┼──────────────────────────────────┼───────────────────────────────┤
codelabz-emulator-1  | │ Pub/Sub        │ 0.0.0.0:8085                     │ n/a                           │
codelabz-emulator-1  | ├────────────────┼──────────────────────────────────┼───────────────────────────────┤
codelabz-emulator-1  | │ Storage        │ 0.0.0.0:9199                     │ http://0.0.0.0:4000/storage   │
codelabz-emulator-1  | └────────────────┴──────────────────────────────────┴───────────────────────────────┘
codelabz-emulator-1  |   Emulator Hub running at localhost:4400
codelabz-emulator-1  |   Other reserved ports: 4500
codelabz-emulator-1  |
codelabz-emulator-1  | Issues? Report them at https://github.com/firebase/firebase-tools/issues and attach the *-debug.log files.
codelabz-emulator-1  |

these are the errors that I am facing while running the docker-compose up command I don't why node_modules is not getting iunstalled

**Comment by manzil-infinity180** [comment]
> codelabz-app-1 | codelabz-app-1 | > codelabz@0.0.0 dev /app codelabz-app-1 | > vite codelabz-app-1 | codelabz-app-1 | sh: 1: vite: not found codelabz-app-1 | npm ERR! code ELIFECYCLE codelabz-app-1 | npm ERR! syscall spawn codelabz-app-1 | npm ERR! file sh codelabz-app-1 | npm ERR! errno ENOENT codelabz-app-1 | npm ERR! codelabz@0.0.0 dev: `vite` codelabz-app-1 | npm ERR! spawn ENOENT codelabz-app-1 | npm ERR! codelabz-app-1 | npm ERR! Failed at the codelabz@0.0.0 dev script. codelabz-app-1 | npm ERR! This is probably not a problem with npm. There is likely additional logging output above. codelabz-app-1 | npm WARN Local package.json exists, but node_modules missing, did you mean to install? codelabz-app-1 | codelabz-app-1 | npm ERR! A complete log of this run can be found in: codelabz-app-1 | npm ERR! /root/.npm/_logs/2023-10-22T01_37_34_503Z-debug.log codelabz-app-1 exited with code 1 codelabz-emulator-1 | ⚠ emulators: You are not currently authenticated so some features may not work correctly. Please run firebase login to authenticate the CLI. codelabz-emulator-1 | i emulators: Starting emulators: auth, functions, firestore, database, pubsub, storage codelabz-emulator-1 | i emulators: Detected demo project ID "demo-sampark", emulated services will use a demo configuration and attempts to access non-emulated services for this project will fail. codelabz-emulator-1 | ⚠ emulators: It seems that you are running multiple instances of the emulator suite for project demo-sampark. This may result in unexpected behavior. codelabz-emulator-1 | ⚠ emulators: Could not find import/export metadata file, skipping data import! codelabz-emulator-1 | ⚠ functions: The functions emulator is configured but there is no functions source directory. Have you run firebase init functions? codelabz-emulator-1 | ⚠ firestore: Did not find a Cloud Firestore rules file specified in a firebase.json config file. codelabz-emulator-1 | ⚠ firestore: The emulator will default to allowing all reads and writes. Learn more about this option: https://firebase.google.com/docs/emulator-suite/install_and_configure#security_rules_configuration. codelabz-emulator-1 | i firestore: Firestore Emulator logging to firestore-debug.log codelabz-emulator-1 | ⚠ database: Did not find a Realtime Database rules file specified in a firebase.json config file. The emulator will default to allowing all reads and writes. Learn more about this option: https://firebase.google.com/docs/emulator-suite/install_and_configure#security_rules_configuration. codelabz-emulator-1 | i database: Database Emulator logging to database-debug.log codelabz-emulator-1 | i pubsub: Pub/Sub Emulator logging to pubsub-debug.log codelabz-emulator-1 | i ui: Emulator UI logging to ui-debug.log codelabz-emulator-1 | codelabz-emulator-1 | ┌─────────────────────────────────────────────────────────────┐ codelabz-emulator-1 | │ ✔ All emulators ready! It is now safe to connect your app. │ codelabz-emulator-1 | │ i View Emulator UI at http://0.0.0.0:4000 │ codelabz-emulator-1 | └─────────────────────────────────────────────────────────────┘ codelabz-emulator-1 | codelabz-emulator-1 | ┌────────────────┬──────────────────────────────────┬───────────────────────────────┐ codelabz-emulator-1 | │ Emulator │ Host:Port │ View in Emulator UI │ codelabz-emulator-1 | ├────────────────┼──────────────────────────────────┼───────────────────────────────┤ codelabz-emulator-1 | │ Authentication │ 0.0.0.0:9099 │ http://0.0.0.0:4000/auth │ codelabz-emulator-1 | ├────────────────┼──────────────────────────────────┼───────────────────────────────┤ codelabz-emulator-1 | │ Functions │ Failed to initialize (see above) │ │ codelabz-emulator-1 | ├────────────────┼──────────────────────────────────┼───────────────────────────────┤ codelabz-emulator-1 | │ Firestore │ 0.0.0.0:8080 │ http://0.0.0.0:4000/firestore │ codelabz-emulator-1 | ├────────────────┼──────────────────────────────────┼───────────────────────────────┤ codelabz-emulator-1 | │ Database │ 0.0.0.0:9000 │ http://0.0.0.0:4000/database │ codelabz-emulator-1 | ├────────────────┼──────────────────────────────────┼───────────────────────────────┤ codelabz-emulator-1 | │ Pub/Sub │ 0.0.0.0:8085 │ n/a │ codelabz-emulator-1 | ├────────────────┼──────────────────────────────────┼───────────────────────────────┤ codelabz-emulator-1 | │ Storage │ 0.0.0.0:9199 │ http://0.0.0.0:4000/storage │ codelabz-emulator-1 | └────────────────┴──────────────────────────────────┴───────────────────────────────┘ codelabz-emulator-1 | Emulator Hub running at localhost:4400 codelabz-emulator-1 | Other reserved ports: 4500 codelabz-emulator-1 | codelabz-emulator-1 | Issues? Report them at https://github.com/firebase/firebase-tools/issues and attach the *-debug.log files. codelabz-emulator-1 |
> 
> these are the errors that I am facing while running the docker-compose up command I don't why node_modules is not getting iunstalled

setup our emulator first
